Do not update id on fetch (fixes #1)

gagath opened the pull request:
The `_update_local_ids()` function will try to update the `mydomain.org_ovh-redirections.yaml` file (with no dot) but the fetch function will create the `.mydomain.org_ovh-redirections.yaml` (with a dot) so thats why there is an error.

Wait for checkout in order to update local ids should be enough.

gagath committed:
Do not update IDs if local database not created

neomilium commented:
Updating IDs is required to have a consistent output after 'commit' : as IDs are generated on server side, you need to re-fetch data to your local cache (.mydomain.tld.yaml) and update IDs in your working file (mydomain.tld.yaml).
So, you need to make some improvements to support this case.

Thanks !

gagath committed:
Improved info in README between fetch and checkout

gagath commented:
I did not saw that the `fetch` command was called after each `commit`.

I have changed the behavior of the fetch method: the `_update_local_ids` function will now be called on every fetch expect if the local database was not found. The first call to `fetch` now does not fail.

This questions the error handling at line 116, since not having the local database *is a normal state* according to the README, between the `fetch` and the `checkout` calls.

neomilium commented on ovh-redirections:
```diff
@@ -71,15 +71,27 @@ def fetch():
+    if os.path.isfile(local_db):
+        _update_local_ids()
+    else:
+        print("Local database not found, skipping update local IDs...")
```
You should remove this displayed text about "not updating ID" : its more a debug message than user message.

neomilium commented:
Great, thanks !

Could you squash your commits that are related to fixing ID on first fetch, and keep only 2 commits (one for fixing, one for README improvement) ?

gagath commented:
I have reworked the pull request in two commits. In the diff we can see two differences:
- Do not update IDs if local database is missing
- Update local IDs on checkout

The first point is correct, but I am not sure about the second one.

neomilium commented on ovh-redirections:
```diff
@@ -96,6 +106,7 @@ def _update_local_ids():
 
 def checkout():
     shutil.copyfile(remote_db, local_db)
+    _update_local_ids()
```
As the file is copied, why updating IDs ?

gagath commented:
Updating the local IDs during the `checkout` was not useful indeed. I have removed it and rebased the pull request in two commits as before: one for the code fix and one for the README.

neomilium commented:
Thanks !
